src/config-schema.js

```diff
@@ -360,13 +360,14 @@ const configSchema = {
       useTreeSitterParsers: {
         type: 'boolean',
         default: true,
+        title: 'Use Tree-sitter Parsers',
         description: 'Use Tree-sitter parsers for supported languages.'
       },
-      useExperimentalModernTreeSitter: {
+      useLegacyTreeSitter: {
         type: 'boolean',
         default: false,
-        title: 'Use Modern Tree-Sitter Implementation',
-        description: 'Experimental: Use the new query-file-based Tree-sitter system instead of the legacy system from Atom. (This system will eventually replace the legacy system.) Has no effect unless "Use Tree Sitter Parsers" is also checked.'
+        title: 'Use Legacy Tree-sitter Implementation',
+        description: 'Opt into the legacy Atom Tree-sitter system instead of the modern system added by Pulsar. (This legacy system will soon be removed.) Has no effect unless “Use Tree-sitter Parsers” is also checked.'
```

Comment:
Suggestion is to change the wording to say that this is planned, not inevitable... since we're still finishing qualifying that modern tree-sitter works up to our standards given the wider testing enabling it by default will bring, and as the more likely reason it won't go away right away: we probably won't remove it until the newer Electron builds are _also_ qualified as good enough, which may take some additional time beyond the new tree-sitter system being up to snuff.

```suggestion
        description: 'Opt into the legacy Atom Tree-sitter system instead of the modern system added by Pulsar. (We plan to remove this legacy system soon.) Has no effect unless “Use Tree-sitter Parsers” is also checked.'
```

My 2 cents. If you want to emphasize the point more strongly as this original wording did, then there may be some merit in that, I suppose.